compiler/src/yul/runtime/functions.rs

```diff
@@ -24,6 +24,7 @@ pub fn avail() -> yul::Statement {
     function_definition! {
         function avail() -> ptr {
             (ptr := mload(0x00))
+            (if (eq(ptr, 0x00)) { (ptr := 0x20) })
```

Comment:
There was a bug here where `avail()` was returning `0x00` as the lowest available memory slot, which is reserved for storing the lowest available memory slot, causing us to set the lowest available memory slot to a very large number and triggering an out-of-gas error.